test/boost/mutation_test.cc

```diff
@@ -2942,10 +2942,9 @@ void run_compaction_data_stream_split_test(const schema& schema, gc_clock::time_
 } // anonymous namespace
```

Comment:
changelog: why should we stop taking partitioner in constructor?

Reply:
added more explanation.